### PR TITLE
update configuration page for `enable-enum-length-limit`

### PR DESCRIPTION
**Pull request by bb7133**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Added the description for a new configuration entry for the 'tidb-configuration-file'.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/19109

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by zimulala — COMMENTED** [review]
/LGTM

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -159,6 +159,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：false
 + 当使用 LVS 为 TiDB 做负载均衡时，可开启此配置项。这是因为 [LVS 的 TOA 模块](https://github.com/alibaba/LVS/tree/master/kernel/net/toa)可以通过 TCP4 协议从 TCP 头部信息中解析出客户端的真实 IP。
 
+### `enable-enum-length-limit` <span class="version-mark">从 v5.0 版本开始引入</span>
+
++ 是否限制 `ENUM` 和 `SET` 两种类型中，单个元素的最大长度
```
Comment (TomShawn) [inline tidb-configuration-file.md:164]:
```suggestion
+ 是否限制单个 `ENUM` 元素和单个 `SET` 元素的最大长度
```

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -159,6 +159,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：true
++ 当为 true 时，`ENUM` 和 `SET` 单个元素最大长度为 255 个字符（[与 MySQL 8 兼容](https://dev.mysql.com/doc/refman/8.0/en/string-type-syntax.html)）；当为 false 时，不对单个元素长度进行限制（与 TiDB v5.0 之前版本兼容）。
```
Comment (TomShawn) [inline tidb-configuration-file.md:166]:
```suggestion
+ 当该配置项值为 `true` 时，`ENUM` 和 `SET` 单个元素的最大长度为 255 个字符，[与 MySQL 8 兼容](https://dev.mysql.com/doc/refman/8.0/en/string-type-syntax.html)；当该配置项值为 `false` 时，不对单个元素的长度进行限制，与 TiDB v5.0 之前的版本兼容。
```

**Comment by bb7133** [comment]
Updated PTAL @TomShawn

**Pull request merged**